### PR TITLE
feat(autofix): Stream output in Autofix agent

### PR DESCRIPTION
**Pull request by roaga**
Replaces Autofix agent LLM calls with streaming. Saves the output as it comes in the step object in the run state. All other functionality is identical.

I added a function for streaming with OpenAI as well, and tested that it works, but we don't use it at the moment. Just future proofing.

### CODE REVIEW

File: src/seer/automation/autofix/components/root_cause/component.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def invoke(
                     for i, item in enumerate(cause_model.code_context):
                         item.id = i
                         # Find line range for the snippet
-                        if item.snippet.file_path and item.snippet.snippet:
```
Comment (roaga) [inline src/seer/automation/autofix/components/root_cause/component.py:131]: tacking on a fix for an unrelated bug

File: src/seer/automation/autofix/components/root_cause/component.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ def invoke(
...
+                        if item.snippet and item.snippet.file_path and item.snippet.snippet:
```
Comment (roaga) [inline src/seer/automation/autofix/components/root_cause/component.py:131]: fix for unrelated bug

**Review by trillville — APPROVED** [review]

File: src/seer/automation/autofix/autofix_agent.py [hunk 3]
```diff
@@ -63,6 +69,49 @@ def _check_prompt_for_help(self, run_config: RunConfig):
                 "You're taking a while. If you need help, ask me a concrete question using the tool provided."
             )
 
+    def get_completion(self, run_config: RunConfig):
```
Comment (jennmueng) [inline src/seer/automation/autofix/autofix_agent.py:72]: Little comment for cleanliness so folks grok right away why this method is special for the autofix agent

```suggestion
    def get_completion(self, run_config: RunConfig):
    """
    Streams the preliminary output to the current step and only returns when output is complete
    """
```

**Review by jennmueng — APPROVED** [review]
LGTM, well tested! One little suggestion

**Pull request merged**